File: TripKitAndroid/src/main/java/com/skedgo/tripkit/LocationInfoApi.java
package com.skedgo.tripkit;

import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.Url;
import io.reactivex.Observable;

interface LocationInfoApi {
  @GET Observable<LocationInfo> fetchLocationInfoAsync(
      @Url String url,
      @Query("lat") double lat,
      @Query("lng") double lng
  );
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/LocationInfoApi.kt
package com.skedgo.tripkit

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Url

interface LocationInfoApi {
    @GET
    fun fetchLocationInfoAsync(
        @Url url: String,
        @Query("lat") lat: Double?,
        @Query("lng") lng: Double?
    ): Observable<LocationInfo>
}